### ISSUE TITLE
Reduce number of autocomplete items on smaller viewports?

### ISSUE THREAD

**Issue by AdamChrimes** [nhsuk/nhsuk-frontend#598]
## Improvement suggestion

It has been suggested we could reduce the number of the Header autocomplete items on smaller viewports to improve web performance, fit better within the viewport and potentially reduce CPU usage on mobile devices.

[Header autocomplete JavaScript](https://github.com/nhsuk/nhsuk-frontend/blob/master/packages/components/header/headerAutoComplete.js#L28)

https://twitter.com/russhuq/status/1245538444622860288

**Comment by davidhunter08** [comment]
Samsung Galaxy 10 plus:

![Screenshot_20200402-083006_Chrome](https://user-images.githubusercontent.com/14331000/78227867-d807f180-74c5-11ea-98af-9e7623fcb4fd.jpg)

**Comment by AdamChrimes** [comment]
iPhone X:

![47919E58-D898-4EA7-AC03-6F130B1477EB](https://user-images.githubusercontent.com/11615501/78228112-392fc500-74c6-11ea-956b-27e0b80fa6cd.png)

without keyboard 

![3AB0BCB3-C460-45AD-90B0-85D3E97B38A1](https://user-images.githubusercontent.com/11615501/78228137-42209680-74c6-11ea-890e-73c467c16875.png)

**Comment by Nooshu** [comment]
OnePlus 6T with keyboard:
![6t-with](https://user-images.githubusercontent.com/1223960/78231703-1f44b100-74cb-11ea-967f-3b1a5fcd9229.jpg)

OnePlus 6T without keyboard:
![6t-without](https://user-images.githubusercontent.com/1223960/78231732-28358280-74cb-11ea-8683-f0d0de964f8c.jpg)

**Comment by Nooshu** [comment]
Alcatel 1 ([budget Android device](https://www.tescomobile.com/shop/pay-as-you-go/alcatel/1)).

With keyboard:
![alcatel-1-with](https://user-images.githubusercontent.com/1223960/78231987-86626580-74cb-11ea-95c3-3e9dd006fcf5.jpeg)

Without keyboard:
![alcatel-1-without](https://user-images.githubusercontent.com/1223960/78231996-8bbfb000-74cb-11ea-81e6-dc32066fc75c.jpeg)

**Comment by mcheung-nhs** [comment]
A quick hack without having to change the javascript would be to hide the last few based on screen width using CSS:

```
@include mq($until: tablet) {
  .autocomplete__option:nth-child(n+6) {
    display: none;
  }
}
```
This would show the first 5 and hide the last 5

**Comment by AdamChrimes** [comment]
We're looking at changing the search provider which should result in more relevant search terms, which might mean we don't have to show as many results as we do now.

@pflynny this might be of interest for the search changes

**Comment by pflynny** [comment]
I'd seen the performance issue raised before, it's really useful information – we'll be looking at the design in the next couple of months, during that time we'll make sure that it is also as performant as possible. As we progress and test it I'll keep you posted.

**Comment by AdamChrimes** [comment]
This will be fixed in release 5.0.0 (next week)
